src/registrar/views/domain.py

```diff
@@ -221,6 +241,300 @@ def form_valid(self, formset):
         return super().form_valid(formset)
 
 
+class DomainDNSSECView(DomainPermissionView, FormMixin):
+
+    """Domain DNSSEC editing view."""
+
+    template_name = "domain_dnssec.html"
+    form_class = DomainDnssecForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        """The initial value for the form (which is a formset here)."""
+        self.domain = self.get_object()
+
+        has_dnssec_records = self.domain.dnssecdata is not None
+
+        # Create HTML for the modal button
+        modal_button = (
+            '<button type="submit" '
+            'class="usa-button" '
+            'name="disable_dnssec">Disable DNSSEC</button>'
+        )
+
+        context["modal_button"] = modal_button
+        context["has_dnssec_records"] = has_dnssec_records
+        context["dnssec_enabled"] = self.request.session.pop("dnssec_enabled", False)
+
+        return context
+
+    def get_success_url(self):
+        """Redirect to the DNSSEC page for the domain."""
+        return reverse("domain-dns-dnssec", kwargs={"pk": self.domain.pk})
+
+    def post(self, request, *args, **kwargs):
+        """Form submission posts to this view."""
+        self.domain = self.get_object()
+        form = self.get_form()
+        if form.is_valid():
+            if "disable_dnssec" in request.POST:
+                try:
+                    self.domain.dnssecdata = {}
+                except RegistryError as err:
+                    errmsg = "Error removing existing DNSSEC record(s)."
+                    logger.error(errmsg + ": " + err)
+                    messages.error(self.request, errmsg)
+                request.session["dnssec_ds_confirmed"] = False
+                request.session["dnssec_key_confirmed"] = False
+            elif "enable_dnssec" in request.POST:
+                request.session["dnssec_enabled"] = True
+                request.session["dnssec_ds_confirmed"] = False
+                request.session["dnssec_key_confirmed"] = False
+
+        return self.form_valid(form)
+
+
+class DomainDsdataView(DomainPermissionView, FormMixin):
```

Comment:
```suggestion
class DomainDsDataView(DomainPermissionView, FormMixin):
```
Total suggestion, just looks like the D should be capitalized here